src/_enums.h

```diff
@@ -0,0 +1,80 @@
+#ifndef MPL_ENUMS_H
+#define MPL_ENUMS_H
+
+#include <pybind11/pybind11.h>
+
+// Extension for pybind11: Pythonic enums.
+
+// a1 includes the opening brace and a2 the closing brace.
+// This definition is compatible with older compiler versions compared to
+// #define P11X_ENUM_TYPE(...) decltype(std::map{std::pair __VA_ARGS__})::mapped_type
+#define P11X_ENUM_TYPE(a1, a2, ...) decltype(std::pair a1, a2)::second_type
+
+#define P11X_CAT2(a, b) a##b
+#define P11X_CAT(a, b) P11X_CAT2(a, b)
+
+namespace p11x {
+  namespace {
+    namespace py = pybind11;
+
+    // Holder is (py_base_cls, [(name, value), ...]) before module init;
+    // converted to the Python class object after init.
+    auto enums = std::unordered_map<std::string, py::object>{};
+
+    auto bind_enums(py::module mod) -> void
+    {
+      for (auto& [py_name, spec]: enums) {
+        auto const& [py_base_cls, pairs] =
+          spec.cast<std::pair<std::string, py::object>>();
+        mod.attr(py::cast(py_name)) = spec =
+          py::module::import("pydoc").attr("locate")(py_base_cls)(
```

Comment:
You can probably use pkgutil.resolve_name here instead of the older and undocumented pydoc.locate (though both would work).

Reply:
Or should we just hard-code this coming from `enum` (and only provide the class name to the macro)?

Reply:
That's probably fine too, anything works here...

Reply:
OK, I just did the hard-coding then.